### PR TITLE
yii\base\ArrayableTrait::toArray serializes objects through Serializer

### PR DESCRIPTION
**Pull request by patacca**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #17889 

It fixed the issue but I don't know if it is ok to use Serializer in this class

### CODE REVIEW

File: framework/base/ArrayableTrait.php [hunk 1]
```diff
@@ -11,6 +11,7 @@
 use yii\helpers\ArrayHelper;
 use yii\web\Link;
 use yii\web\Linkable;
+use yii\rest\Serializer;
```
Comment (samdark) [inline framework/base/ArrayableTrait.php]: Overall that doesn't look too good. Is there another way to solve it?

Comment (patacca) [reply framework/base/ArrayableTrait.php]: See this https://github.com/yiisoft/yii2/issues/17889#issuecomment-590565449

File: framework/base/ArrayableTrait.php [hunk 2]
```diff
@@ -121,6 +122,8 @@ public function extraFields()
      */
     public function toArray(array $fields = [], array $expand = [], $recursive = true)
     {
+        $serializer = new Serializer();
```
Comment (romkatsu) [inline framework/base/ArrayableTrait.php]: Duplication. Maybe it's better to put it in the method?

```php
private function createSerializer() {
   $serializer = new Serializer();
   $serializer->preserveKeys = true;
   return $serializer;
}

```

**Review by romkatsu — CHANGES_REQUESTED** [review]

File: framework/base/ArrayableTrait.php (hunk 2 above)
Comment (patacca) [reply framework/base/ArrayableTrait.php]: I think this isn't going to be merged. Objects serializable should just be treated as non-serializable
See this https://github.com/yiisoft/yii2/issues/17889#issuecomment-591518449

**Pull request closed without merging**